src/Bicep.Cli/Services/ParametersWriter.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Emit;
+using Bicep.Core.Exceptions;
+using Bicep.Core.Semantics;
+using Newtonsoft.Json;
+using System;
+using System.IO;
+
+namespace Bicep.Cli.Services
+{
+    public class ParametersWriter
+    {
+        private readonly InvocationContext invocationContext;
+
+        public ParametersWriter(InvocationContext invocationContext)
+        {
+            this.invocationContext = invocationContext;
+        }
+
+        public EmitResult ToFile(ParamsSemanticModel paramSemanticModel, string outputPath)
+        {
+            using var fileStream = CreateFileStream(outputPath);
+            return new ParametersEmitter(paramSemanticModel, invocationContext.EmitterSettings).EmitParamsFile(fileStream);
+        }
+
+        public EmitResult ToStdout(ParamsSemanticModel paramSemanticModel)
+        {
+            using var writer = new JsonTextWriter(invocationContext.OutputWriter)
+            {
+                Formatting = Formatting.Indented
+            };new ParametersEmitter(paramSemanticModel, invocationContext.EmitterSettings).EmitParamsFile(writer);
```

Comment:
>new [](http://example.com/codeflow?start=14&length=3)

nit - Since it's a different statement, it should start on a new line. It makes it easier to read.